src/Spid/Saml.php

```diff
@@ -219,6 +220,10 @@ public function isAuthenticated() : bool
             $session = new Session($_SESSION['spidSession']);
             if ($session->isValid()) {
                 $this->session = $session;
+                if (isset($this->settings['database'])) {
+                    $db = new Db($idp);
```

Comment:
This is a little bit too hardcoded. Can we think about something like a storage handler? We can provide a reference implementation that stores request and response to db, but others can have different requirements.
For example the SPID module for Drupal can provide an handler that uses the database connection already in place

Reply:
What about make accessible the request and the response from login.php and acs.php files? So everyone can implement the db connection as they want.

Reply:
I mean, maybe it's not a database at all, it may be a filesystem, a Redis instance, etc. I prefer to have a generic storage handler. Then in the library we can have a databaseStorageHandler that implements what you're doing

Reply:
> I mean, maybe it's not a database at all, it may be a filesystem, a Redis instance, etc. I prefer to have a generic storage handler. Then in the library we can have a databaseStorageHandler that implements what you're doing

Ok Luca, this have sense. Could you accept only the modifies in Saml.php, Idp.php and BaseResponse.php? I have only added get functions. So, in the meanwhile, I can manage the db connection outside the library. If you get me your ok, I can make another commit with only this modifies

Reply:
Of course, that part is OK for me :-)

Reply:
> Of course, that part is OK for me :-)

Done